src/ssh_fmt_plug.c

```diff
@@ -302,6 +302,7 @@ static void *get_salt(char *ciphertext)
 					ERR_print_errors_fp(stderr);
 					error();
 				}
+				return NULL;
```

Comment:
Wouldn't that return NULL skip the OPENSSL_free..., may be we need a goto for error handling/return?